src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/SelectAllText/SelectAllTextBehavior.ios.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UIKit;
+using ObjCRuntime;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+/// <summary>
+/// A behavior that selects all text when the view is focused.
+/// </summary>
+public class SelectAllTextBehavior : PlatformBehavior<InputView>
+{
+	InputView? element;
+	UIView? control;
+
+	/// <inheritdoc/>
+	protected override void OnAttachedTo(InputView bindable, UIView platformView)
+	{
+		element = bindable;
+		control = platformView;
+		ApplyBehaviorToControl(true, platformView);
+	}
+	
+	bool ApplyBehaviorToControl<T>(bool apply, T platformView)
+	{
+		return platformView switch
+		{
+			UITextField textField => ApplyToUITextField(textField, apply),
+			UITextView => ApplyToUITextView(apply),
+			_ => throw new NotSupportedException($"Control of type: {platformView?.GetType()?.Name} is not supported by this effect.")
+		};
+	}
+
+	bool ApplyToUITextField(UITextField textField, bool apply)
+	{
+		if (textField is null)
+		{
+			return false;
+		}
+
+		if (apply)
+		{
+			textField.EditingDidBegin += OnEditingDidBegin;
+		}
+		else
+		{
+			textField.EditingDidBegin -= OnEditingDidBegin;
+		}
+
+		return true;
+	}
+
+	void OnEditingDidBegin(object? sender, EventArgs e)
+	{
+		if (sender is not UITextField textfield)
+		{
+			return;
+		}
```

Comment:
Should this throw an `ArgumentException` or an `InvalidOperationException`?


Is any other type subscribing to this event handler that isn't a `UITextField`? If not, let's throw an Exception.


Or, better, just cast it:

```suggestion
		var texfield = (UITextField)sender;
```

Reply:
same as before, I just maintained the previous behavior, as you can see [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/22b1d4ccf45555c2ed89ecb94e9ff93a5cf61ab3/src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.ios.cs#L73-L74). Do you think that we should throw an exception?

If I'm not wrong, throwing an exception at this level could crash the app, not sure if the behavior is too important to the point of being able to break the application

Reply:
Resolved in another discussion